Mathlib/Analysis/Meromorphic/Order.lean

```diff
@@ -165,6 +173,90 @@ theorem order_inv {f : 𝕜 → 𝕜} {z₀ : 𝕜} (hf : MeromorphicAt f z₀)
   simp only [Pi.inv_apply, h₁a h₂a, smul_eq_mul, mul_inv_rev, zpow_neg]
   ring
 
+/--
+The order of a sum at least the minimum of the orders of the summands.
+-/
+theorem order_add {f₁ f₂ : 𝕜 → E} {x : 𝕜} (hf₁ : MeromorphicAt f₁ x) (hf₂ : MeromorphicAt f₂ x) :
+    min hf₁.order hf₂.order ≤ (hf₁.add hf₂).order := by
+  -- Handle the trivial cases where one of the orders equals ⊤
+  by_cases h₂f₁: hf₁.order = ⊤
+  · simp only [h₂f₁, le_top, inf_of_le_right]
+    rw [(hf₁.add hf₂).order_congr]
+    filter_upwards [hf₁.order_eq_top_iff.1 h₂f₁]
+    simp
+  by_cases h₂f₂: hf₂.order = ⊤
+  · simp only [h₂f₂, le_top, inf_of_le_left]
+    rw [(hf₁.add hf₂).order_congr]
+    filter_upwards [hf₂.order_eq_top_iff.1 h₂f₂]
+    simp
+  -- General case
+  lift hf₁.order to ℤ using h₂f₁ with n₁ hn₁
+  lift hf₂.order to ℤ using h₂f₂ with n₂ hn₂
+  obtain ⟨g₁, h₁g₁, h₂g₁, h₃g₁⟩ := (hf₁.order_eq_int_iff n₁).1 hn₁.symm
+  obtain ⟨g₂, h₁g₂, h₂g₂, h₃g₂⟩ := (hf₂.order_eq_int_iff n₂).1 hn₂.symm
+  let n := min n₁ n₂
+  let g := (fun z ↦ (z - x) ^ (n₁ - n)) • g₁ +  (fun z ↦ (z - x) ^ (n₂ - n)) • g₂
+  have h₁g : AnalyticAt 𝕜 g x := by
+    apply AnalyticAt.add
+    apply (AnalyticAt.zpow_nonneg (by fun_prop) (sub_nonneg.2 (Int.min_le_left n₁ n₂))).smul h₁g₁
+    apply (AnalyticAt.zpow_nonneg (by fun_prop) (sub_nonneg.2 (Int.min_le_right n₁ n₂))).smul h₁g₂
+  have : f₁ + f₂ =ᶠ[𝓝[≠] x] ((· - x) ^ n) • g := by
+    filter_upwards [h₃g₁, h₃g₂, (self_mem_nhdsWithin : {x}ᶜ ∈ 𝓝[≠] x)]
+    unfold g
+    simp_all [smul_add, ← smul_assoc, ← zpow_add', sub_ne_zero]
+  have t₀ : MeromorphicAt ((·  - x) ^ n) x := by fun_prop
+  have t₁ : t₀.order = n := by
+    rw [t₀.order_eq_int_iff]
+    use 1, analyticAt_const
+    simp
```

Comment:
```suggestion
  have t₁ : t₀.order = n := (t₀.order_eq_int_iff _).mpr ⟨1, analyticAt_const, by simp⟩
```
This suggests that the `n` argument to `MeromorphicAt.order_eq_int_iff` should be made implicit – the general rule is that arguments to `iff` statements should be implicit if they are inferrable from either side of the `iff`.